### PR TITLE
Fix `ReadonlyReplica` metric for all cases

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `ReadonlyReplica` metric for some cases (e.g. when a table cannot be initialized because of difference in local and Keeper data).

cc @tavplubix
IMO it became too complex and hard to follow the state of the metric for a specific table (with different booleans and checks done for it)
so I introduced `is_readonly_metric_set` flag to make it much easier to reason about the logic.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 2ea44adefdced4f9e5bd5ee897748f89d780db2f with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57267/2ea44adefdced4f9e5bd5ee897748f89d780db2f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7006963281">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57267/2ea44adefdced4f9e5bd5ee897748f89d780db2f/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by tavplubix — APPROVED** [review]

**Review by tavplubix — APPROVED** [review]
